CMake fails for iOS: WASMTIME_TARGET check is wrong

Right now if you try and build for iOS:

```bash
cmake -S crates/c-api -B target/c-api --install-prefix "$(pwd)/artifacts" -DWASMTIME_TARGET=aarch64-apple-ios -DWASMTIME_FEATURES="parallel-compilation,pooling-allocator,cranelift,demangle,profiling,coredump,addr2line,winch,pulley"

cmake --build target/c-api
cmake --install target/c-api
```

The cmake install fails:

```
CMake Error at target/c-api/cmake_install.cmake:45 (file):
  file INSTALL cannot find
  "/Users/seph/3rdparty/wasmtime/crates/c-api/../../target/aarch64-apple-ios/release/libwasmtime.so":
  No such file or directory.
```

The problem is this line in CMakeLists.txt:

```cmake
if(WASMTIME_TARGET MATCHES "darwin")
  set(WASMTIME_SHARED_FILES libwasmtime.dylib)
  set(WASMTIME_STATIC_FILES libwasmtime.a)
elseif(...)
```

This case should also match for ios, and other apple targets. Changing it to this:

```cmake
if(WASMTIME_TARGET MATCHES "apple")
```

Fixes the problem.

--- a/crates/c-api/CMakeLists.txt
+++ b/crates/c-api/CMakeLists.txt
@@ -34,7 +34,7 @@ endif()
 
 set(WASMTIME_TARGET_DIR ${CMAKE_CURRENT_SOURCE_DIR}/../../target/${WASMTIME_TARGET}/${WASMTIME_BUILD_TYPE})
 
-if(WASMTIME_TARGET MATCHES "darwin")
+if(WASMTIME_TARGET MATCHES "apple")
   set(WASMTIME_SHARED_FILES libwasmtime.dylib)
   set(WASMTIME_STATIC_FILES libwasmtime.a)
 elseif(WASMTIME_TARGET MATCHES "windows-gnu")

